### ISSUE TITLE
Options for Unicode variable name

### ISSUE THREAD

**Issue by minogram** [codingseb/ExpressionEvaluator#63]
Hi 
I'd like to say your work is quite a nice and useful to me. Thank you.

I want to use "Unicode variable name" like standard c#. (for example Korean, Chinese ...), 
so I modified your code as below, and then it works fine for Korean.

```
// changed line
const string diactiticsKeywordsRegexPattern = @"\p{L}_";
```

Example

```
// english
one = 1;
two = 2;
three = one + two;

// korean
하나 = 1;
둘 = 2;
셋 = 하나 + 둘;
```

I hope, you can make a option, like "OptionsUnicodeVariableNameActive".

Thank you!

**Comment by codingseb** [comment]
Hello @minogram, thanks for opening this issue.
And thanks also for providing inputs for the solution.

As my first language is french and I only know a bit of english as second language, I am not used to asian languages with other type of characters (Maybe one day). So I have a question for you.

It seems that your solution also allow all defined diactitics so do you think that we really need an option ?

In my opinion your solution looks better than the actual one and allow by default a larger set of letters character in more languages for variables, classes, methods and properties names,. But I don't now if I miss something. Also I was not aware of \p{L} stuffs in Regex until now.

When you program in C# in korean do you usually need to activate a option or change something in your IDE ? or is it supported by default by the language ?

I think that I could simply publish a new version with your solution as you provided it. What do you think ?

**Comment by codingseb** [comment]
OK I did what I should have tried directly

I tried your example directly in C# in my Visual Studio 2019.

```c#
using System;

namespace UnicodeCodingTest
{
    class Program
    {
        static void Main(string[] args)
        {
            var 하나 = 1;
            var 둘 = 2;
            var 셋 = 하나 + 둘;

            Console.WriteLine(셋);
        }
    }
}
```

And it works without doing anything more. So we can say that C# support Unicode language naming by default.
So for me no need to add an option I will integrate your solution directly in next release to support Unicode naming by default.

**Comment by codingseb** [comment]
I just published version 1.4.14.0 that contains your solution. Thanks again for your solution.

**Comment by minogram** [comment]
Dear @codingseb 
Sorry for my last reply. You did the test already. 
Thank you very much on behalf of all asian coder. Have a nice day.
